### PR TITLE
Updated all modules to work with Angular 5

### PR DESCRIPTION
**Pull request by mbejda**
Updated all modules to work with Angular 5 and latest AngularFire 5.0.0-rc.4

### CODE REVIEW

**Comment by mbejda** [comment]
I had to include a test API Key in there otherwise AngularFire barfs errors. Let me know what you think.

File: src/app/app.module.ts [hunk 1]
```diff
@@ -17,12 +17,12 @@ import { AlertsService } from './shared/alerts/alerts.service';
 import { AlertsComponent } from './shared/alerts/alerts.component';
 
 export const firebaseConfig = {
-  apiKey: '',
-  authDomain: '',
-  databaseURL: '',
-  projectId: '',
-  storageBucket: '',
-  messagingSenderId: ''
+    apiKey: "AdIzaSyDZRMQ1axFmRcg5yv1unNFEqnwXTckXGMg44", /// This is a random test key. you need to change this API key to your API key.
```
Comment (pbouda) [inline src/app/app.module.ts]: Let's use single quotes for the settings.

File: src/app/shared/auth.service.ts [hunk 2]
```diff
@@ -94,7 +94,7 @@ export class AuthService {
   setMailAddress(newUser: User) {
     return this.user.first()
     .flatMap(
-      (user) => {
+      (user:any) => {
```
Comment (pbouda) [inline src/app/shared/auth.service.ts:97]: Whitespace missing before the type.

File: package.json [hunk 3]
```diff
@@ -10,43 +10,43 @@
...
-    "@angular/platform-browser-dynamic": "^4.0.0",
-    "@angular/platform-server": "^4.0.0",
-    "@angular/router": "^4.0.0",
-    "@ng-bootstrap/ng-bootstrap": "^1.0.0-alpha.22",
-    "angularfire2": "^4.0.0-rc.0",
-    "core-js": "^2.4.1",
-    "firebase": "^3.9.0",
-    "rxjs": "^5.1.0",
-    "zone.js": "^0.8.4"
+    "@angular/animations": "5.1.2",
```
Comment (pbouda) [inline package.json:15]: Should we still allow to update to minor releases with the '^'? We have the package-lock.json file now to lock versions, so I think it's fine to allow later versions here.

**Review by pbouda — COMMENTED** [review]
Very cool, thanks a lot! I added a few comments, could you have a look at those please?

**Pull request closed without merging**